File: projects/10/ruby/Kiwi.rb
# Kiwi.rb | Defines the Kiwi class which inherits attributes and methods
#   from the walkingBird superclass.
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Daniela Contreras
# Date: 03/30/22
####################################################

require './WalkingBird.rb'

class Kiwi < WalkingBird

##############################################
## Call returns sound of Duck                 #
##Return: Quack! a string                    #
##############################################
    def call
        'kee-wee!'
    end
    
end

File: projects/10/Kiwi.java
/* Kiwi.java provides a Kiwi subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Daniela Contreras
 * Date: 03/30/22
 ******************************************************/

public class Kiwi extends WalkingBird {

    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
     public Kiwi()
     {
        super();
     }
   
     public Kiwi(String name)
     {     
       super(name);
     } 
   
    /* A Kiwi's Call
     * Return: a Kiwi-call ("kee-wee!").
     */

     public String call()
     {
       return "kee-wee!";
     } 
   
   
   }